### ISSUE TITLE
Modularization of sagelib: Break out a separate package sagemath-polyhedra

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#32432]
<div id="comment:0"></div>

We define a superset of the distribution **sagemath-categories** (from #29865), called https://pypi.org/project/sagemath-polyhedra/, which is large enough to work with rational polyhedra, including running most doctests

In:
- `ZZ`, `QQ`, `RDF`
- enough of `sage.modules`, `sage.matrix` to do linear algebra over these rings
- `install_requires` on `pplpy`

Out:
- real fields other than `RDF`
- number fields (we conditionalize out doctests that use them - #32652)
- `sage.symbolic`
- most of `sage.combinat` (we conditionalize out doctests that use combinatorial examples such as associahedra)
- `sage.graphs`
- `sage.plot`

To test:

```
$ ./bootstrap && ./sage -sh -c '(cd pkgs/sagemath-polyhedra && SAGE_NUM_THREADS=16 tox -v -v -v)'
```
To run more of the testsuite afterwards:

```
$ . pkgs/sagemath-polyhedra/.tox/python/bin/activate
(python) $ (cd $(python -c "from sage.env import SAGE_SRC; print(SAGE_SRC)") && sage -tp --environment=sage.all__sagemath_polyhedra sage/geometry)
...
(python) $ deactivate
```

Follow-up steps:
- redo as distribution with dependency on **sagemath-categories**
- wheels for this distribution, for **pplpy** and **PyNormaliz**
- native Windows port

Depends on #32779
Depends on #34855
Depends on #34849
Depends on #32709
Depends on #34849
Depends on #34931
Depends on #34941

CC:  @kliem @tscrim @anneschilling

Component: **refactoring**

Work Issues: **git blame -b --first-parent 32432-base.. src/sage/geometry/polyhedron/base.py**

Author: **Matthias Koeppe**

_Issue created by migration from https://trac.sagemath.org/ticket/32432_

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -6,5 +6,9 @@
 Out:
 - symbolics
 
+To test:
 
+```
+./bootstrap && ./sage -sh -c '(cd pkgs/sagemath-polyhedra && tox -v -v -v)'
+```
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -4,6 +4,7 @@
 - `install_requires` on `pplpy`
 
 Out:
+- real fields
 - symbolics
 
 To test:
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,11 +1,15 @@
 We define a superset of `sagemath-categories` that is large enough to work with rational polyhedra, including running most doctests
 
 In:
+- `ZZ`, `QQ`
 - `install_requires` on `pplpy`
 
 Out:
 - real fields
-- symbolics
+- number fields
+- `sage.symbolic`
+- `sage.combinat`
+- `sage.graphs`
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -14,6 +14,6 @@
 To test:
 
 ```
-./bootstrap && ./sage -sh -c '(cd pkgs/sagemath-polyhedra && tox -v -v -v)'
+./bootstrap && ./sage -sh -c '(cd pkgs/sagemath-polyhedra && SAGE_NUM_THREADS=16 tox -v -v -v)'
 ```
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,14 +1,15 @@
-We define a superset of `sagemath-categories` that is large enough to work with rational polyhedra, including running most doctests
+We define a superset of the distribution **sagemath-categories** (from #29865), called **sagemath-polyhedra** that is large enough to work with rational polyhedra, including running most doctests
 
 In:
 - `ZZ`, `QQ`
+- enough of `sage.modules`, `sage.matrix` to do linear algebra over these rings
 - `install_requires` on `pplpy`
 
 Out:
 - real fields
-- number fields
+- number fields (we conditionalize out doctests that use them - #32652)
 - `sage.symbolic`
-- `sage.combinat`
+- most of `sage.combinat` (we conditionalize out doctests that use combinatorial examples such as associahedra)
 - `sage.graphs`
 
 To test:
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,16 +1,17 @@
 We define a superset of the distribution **sagemath-categories** (from #29865), called **sagemath-polyhedra** that is large enough to work with rational polyhedra, including running most doctests
 
 In:
-- `ZZ`, `QQ`
+- `ZZ`, `QQ`, `RDF`
 - enough of `sage.modules`, `sage.matrix` to do linear algebra over these rings
 - `install_requires` on `pplpy`
 
 Out:
-- real fields
+- real fields other than `RDF`
 - number fields (we conditionalize out doctests that use them - #32652)
 - `sage.symbolic`
 - most of `sage.combinat` (we conditionalize out doctests that use combinatorial examples such as associahedra)
 - `sage.graphs`
+- `sage.plot`
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,4 +1,4 @@
-We define a superset of the distribution **sagemath-categories** (from #29865), called **sagemath-polyhedra** that is large enough to work with rational polyhedra, including running most doctests
+We define a superset of the distribution **sagemath-categories** (from #29865), called https://pypi.org/project/sagemath-polyhedra/, which is large enough to work with rational polyhedra, including running most doctests
 
 In:
 - `ZZ`, `QQ`, `RDF`
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -19,3 +19,8 @@
 ./bootstrap && ./sage -sh -c '(cd pkgs/sagemath-polyhedra && SAGE_NUM_THREADS=16 tox -v -v -v)'
 ```
 
+Follow-up steps:
+- redo as distribution with dependency on **sagemath-categories**
+- wheels for this distribution, for **pplpy** and **PyNormaliz**
+- native Windows port
+
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -16,7 +16,15 @@
 To test:
 
 ```
-./bootstrap && ./sage -sh -c '(cd pkgs/sagemath-polyhedra && SAGE_NUM_THREADS=16 tox -v -v -v)'
+$ ./bootstrap && ./sage -sh -c '(cd pkgs/sagemath-polyhedra && SAGE_NUM_THREADS=16 tox -v -v -v)'
+```
+To run more of the testsuite afterwards:
+
+```
+$ . pkgs/sagemath-polyhedra/.tox/python/bin/activate
+(python) $ sage -tp src/sage/geometry/polyhedron
+...
+(python) $ deactivate
 ```
 
 Follow-up steps:
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -22,7 +22,7 @@
 
 ```
 $ . pkgs/sagemath-polyhedra/.tox/python/bin/activate
-(python) $ sage -tp src/sage/geometry/polyhedron
+(python) $ sage -tp --environment=sage.all__sagemath_polyhedra src/sage/geometry/polyhedron
 ...
 (python) $ deactivate
 ```
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -22,7 +22,7 @@
 
 ```
 $ . pkgs/sagemath-polyhedra/.tox/python/bin/activate
-(python) $ sage -tp --environment=sage.all__sagemath_polyhedra src/sage/geometry/polyhedron
+(python) $ (cd $(python -c "from sage.env import SAGE_SRC; print(SAGE_SRC)") && sage -tp --environment=sage.all__sagemath_polyhedra sage/geometry)
 ...
 (python) $ deactivate
 ```
``````

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:202"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/1c3504fd23fdad4c775ddd025d3da97008f08961"><code>1c3504f</code></a></td><td><code>Merge tag '9.8.beta0' into t/32432/modularization_of_sagelib__break_out_a_separate_package_sagemath_polyhedra</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/9fd58175bed41fcb34eb019bf2793cff9148b0db"><code>9fd5817</code></a></td><td><code>Merge tag '9.7.beta8' into t/33812/refactor_distributions_sage_setup__sagemath__objects_categories__through_sagemath__environment_repl_</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e41e2779c9f5b0d442095e3979d6032c6f2562ff"><code>e41e277</code></a></td><td><code>Merge tag '9.7' into t/33812/refactor_distributions_sage_setup__sagemath__objects_categories__through_sagemath__environment_repl_</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/996461a5e06cef4faa60b646446a62cff3041ede"><code>996461a</code></a></td><td><code>build/pkgs/sagemath_repl/dependencies: Update from pkgs/sagemath-repl/setup.cfg.m4</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/0ffdd57a963b29ec2c681975fc5c7e8682c82b24"><code>0ffdd57</code></a></td><td><code>pkgs/sagemath-repl/setup.cfg.m4: Update python-requires</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/141318f276c07658845e2df41d3255c9c9bbbbc3"><code>141318f</code></a></td><td><code>build/pkgs/sagemath_categories/dependencies: Remove dependencies implied by sagemath_objects dependencies</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/3da18116695c59d26f5243386a303ae04357e412"><code>3da1811</code></a></td><td><code>Wrapping lines of README.rst</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e5d1070e3b851da062e00ead6832ea30cd6e1a8e"><code>e5d1070</code></a></td><td><code>src/sage/misc/lazy_import.pyx: Add comments</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/2b2a226a60f13c51c1a6e47296359278a13376ea"><code>2b2a226</code></a></td><td><code>Merge #33812</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`286fd5e`](https://github.com/sagemath/sagetrac-mirror/commit/286fd5ef6af15f534148a1882cba567c85c1b5f9)** to **[`2b2a226`](https://github.com/sagemath/sagetrac-mirror/commit/2b2a226a60f13c51c1a6e47296359278a13376ea)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`2b2a226`](https://github.com/sagemath/sagetrac-mirror/commit/2b2a226a60f13c51c1a6e47296359278a13376ea)** to **[`b7b0d5f`](https://github.com/sagemath/sagetrac-mirror/commit/b7b0d5f1152613b3e6a974e92e04c186df93eb3f)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:203"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/b7b0d5f1152613b3e6a974e92e04c186df93eb3f"><code>b7b0d5f</code></a></td><td><code>Merge tag '9.8.beta3' into t/32432/modularization_of_sagelib__break_out_a_separate_package_sagemath_polyhedra</code></td></tr></table>

**Comment by mkoeppe** [comment]
Changed dependencies from **#32779, #34195, #33011** to **#32779, #34195, #33011, #34804**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`b7b0d5f`](https://github.com/sagemath/sagetrac-mirror/commit/b7b0d5f1152613b3e6a974e92e04c186df93eb3f)** to **[`f07052d`](https://github.com/sagemath/sagetrac-mirror/commit/f07052d2bbd9b4231ce78dc5e2bace210b087ad6)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:205"></div>

Branch pushed to git repo; I updated commit sha1. Last 10 new commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/8572f1576a544ac533507cbfe09811d47943b518"><code>8572f15</code></a></td><td><code>sage.interfaces.abc.GapElement: New, use it instead of sage.interfaces.gap.is_GapElement</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/adc231ef98032f86f42167a98dd86f3800f1ee9f"><code>adc231e</code></a></td><td><code>remove is_GapElement from number fields code</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/7c67a1089852beca7b5e0b80ad3cb2a319f0ceef"><code>7c67a10</code></a></td><td><code>deprecating is_GapElement</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/6c75e9b0cc5679ee9ea5773ca1586eacf438e378"><code>6c75e9b</code></a></td><td><code>better rst, point at sage.interfaces.abc to use</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e99db3af3a814e9accdd10a915f651aabe9faa91"><code>e99db3a</code></a></td><td><code>src/sage/interfaces/gap.py: Reviewer edits</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/b8024150aa19c47fb893eca9fccb00bcc7c3fd54"><code>b802415</code></a></td><td><code>Merge #34823</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/27c8788aaea7e9e8fd3d4540ce4e56b5242d884f"><code>27c8788</code></a></td><td><code>src/sage/interfaces/abc.py (GpElement): New, use it for isinstance testing</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/106a85b699eb16eb328451fa23c2dee133362a26"><code>106a85b</code></a></td><td><code>src/sage/interfaces/abc.py (SingularElement, Macaulay2Element): New, use for isinstance testing</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/2604174d0ad2a7825d356ce8c219033cbf96eb96"><code>2604174</code></a></td><td><code>src/sage/interfaces/abc.py: Add doc</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/f07052d2bbd9b4231ce78dc5e2bace210b087ad6"><code>f07052d</code></a></td><td><code>Merge #34804</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:206"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c1fa66411dde8ccbcc7099e46d35a9ea8571d888"><code>c1fa664</code></a></td><td><code>pkgs/sagemath-polyhedra/tox.ini: Update</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/36e2e222b9a0c65d1ca86dc386740c14dc330423"><code>36e2e22</code></a></td><td><code>Unmerge #30748</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`f07052d`](https://github.com/sagemath/sagetrac-mirror/commit/f07052d2bbd9b4231ce78dc5e2bace210b087ad6)** to **[`36e2e22`](https://github.com/sagemath/sagetrac-mirror/commit/36e2e222b9a0c65d1ca86dc386740c14dc330423)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:207"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c11efabf09e9651dcf3a919813a218851cc3523a"><code>c11efab</code></a></td><td><code>build/pkgs/sagemath_polyhedra: Update dependencies, add install-requires</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/56fb8928aa6a4dc8b7619a90867db7fad9b16785"><code>56fb892</code></a></td><td><code>src/sage/misc/all__sagemath_categories.py: New</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`36e2e22`](https://github.com/sagemath/sagetrac-mirror/commit/36e2e222b9a0c65d1ca86dc386740c14dc330423)** to **[`56fb892`](https://github.com/sagemath/sagetrac-mirror/commit/56fb8928aa6a4dc8b7619a90867db7fad9b16785)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`56fb892`](https://github.com/sagemath/sagetrac-mirror/commit/56fb8928aa6a4dc8b7619a90867db7fad9b16785)** to **[`ce1e43b`](https://github.com/sagemath/sagetrac-mirror/commit/ce1e43b9c139f595a6eeb2f41fd58fac2e71f82e)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:208"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a3c40980ce16ac6f21a8f17534c4b6b5035eaa5a"><code>a3c4098</code></a></td><td><code>Revert "sage_setup.find.find_python_sources: Do not recurse into subdirectories that are not packages/namespace packages"</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/958d47d17353f20d5695a5dca35416d10a1238b9"><code>958d47d</code></a></td><td><code>pkgs/sagemath-categories/MANIFEST.in.m4: Remove some `__init__` files</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ce1e43b9c139f595a6eeb2f41fd58fac2e71f82e"><code>ce1e43b</code></a></td><td><code>pkgs/sagemath-polyhedra/MANIFEST.in.m4: Remove some `__init__` files</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:209"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/4bcb10fc4723b737bfecf2bf4276671433b96e05"><code>4bcb10f</code></a></td><td><code>Merge tag '9.8.beta5' into t/32432/modularization_of_sagelib__break_out_a_separate_package_sagemath_polyhedra</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`ce1e43b`](https://github.com/sagemath/sagetrac-mirror/commit/ce1e43b9c139f595a6eeb2f41fd58fac2e71f82e)** to **[`4bcb10f`](https://github.com/sagemath/sagetrac-mirror/commit/4bcb10fc4723b737bfecf2bf4276671433b96e05)**

**Comment by mkoeppe** [comment]
Changed dependencies from **#32779, #34195, #33011, #34804** to **#32779, #34195, #33011, #34804, #34855**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:211"></div>

Branch pushed to git repo; I updated commit sha1. Last 10 new commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/00deebede6ae54768e8648bee89038e26d3515f8"><code>00deebe</code></a></td><td><code>pkgs/sagemath-objects/setup.py: Run finder also for sdist, egg_info, dist_info</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/39bd748020fb651e5aeb7640ffe182e493353c79"><code>39bd748</code></a></td><td><code>src/sage/dynamics/arithmetic_dynamics/projective_ds.py: Remove unused import of typing_extensions</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e22943612656b225396e4dea4fae654108b3cf3d"><code>e229436</code></a></td><td><code>Merge #34859</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/2b39b3d2e2b0a8720dfcd764345c03c120e06396"><code>2b39b3d</code></a></td><td><code>Merge #34858</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c59622a42624037d4eba61e353ea07ae00bd55f6"><code>c59622a</code></a></td><td><code>build/pkgs/sagemath*/spkg-src: Use build</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/7fc93d06de9f0e34d2ea2077f1a022750c722235"><code>7fc93d0</code></a></td><td><code>build/pkgs/sagemath*/spkg-src: Use --skip-dependency-check</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/4940db4cf4d32617f23a87e04bddaa01061a7f5f"><code>4940db4</code></a></td><td><code>build/make/Makefile.in: Add targets SPKG-src for script packages</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/573fcfdfba04041260f8e06e54e37eb8a9f85da4"><code>573fcfd</code></a></td><td><code>build/make/Makefile.in: Move targets pypi-sdists, wheel, pypi-wheels here; use SPKG-sdist targets</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/0bd13ac7095789740ce098e23a7b0af4978c621c"><code>0bd13ac</code></a></td><td><code>build/make/Makefile.in: Remove duplicate setting of PYPI_WHEEL_PACKAGES</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/6bf44ecfc4694d595d0e317475a0ecee8b01ddca"><code>6bf44ec</code></a></td><td><code>Merge #34855</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`4bcb10f`](https://github.com/sagemath/sagetrac-mirror/commit/4bcb10fc4723b737bfecf2bf4276671433b96e05)** to **[`6bf44ec`](https://github.com/sagemath/sagetrac-mirror/commit/6bf44ecfc4694d595d0e317475a0ecee8b01ddca)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`6bf44ec`](https://github.com/sagemath/sagetrac-mirror/commit/6bf44ecfc4694d595d0e317475a0ecee8b01ddca)** to **[`702ee8d`](https://github.com/sagemath/sagetrac-mirror/commit/702ee8d5a50971f0022079efeeafb3d522216e65)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:212"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/702ee8d5a50971f0022079efeeafb3d522216e65"><code>702ee8d</code></a></td><td><code>build/pkgs/sagemath*/spkg-src: Use build</code></td></tr></table>

**Comment by mkoeppe** [comment]
Changed dependencies from **#32779, #34195, #33011, #34804, #34855** to **#32779, #34855, #34849, #32709**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:215"></div>

Branch pushed to git repo; I updated commit sha1. Last 10 new commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/449b56bc56bb24a8aa958317463757c6feedf90c"><code>449b56b</code></a></td><td><code>pkgs/sagemath*/tox.ini: Fix for tox 4</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/4a27a3d089dff6f50dde00a68d896e28fed9a9c8"><code>4a27a3d</code></a></td><td><code>Merge #34839</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/95fe89ab0635531f4626a5711343fb47da280374"><code>95fe89a</code></a></td><td><code>build/pkgs/tox/spkg-configure.m4: Check for good tox 3, tox 4 versions separately</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/cc7196abb897a2b3d5bb939e800131b77322dd38"><code>cc7196a</code></a></td><td><code>src/tox.ini: More allowlist_externals for tox 4</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/1999bd6f42480d570deb10f85d6d4219c2936af4"><code>1999bd6</code></a></td><td><code>src/tox.ini: Quoting fix for tox 4</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/071c113fe6f7d4405b4f470c6f3ff02eb8359e3d"><code>071c113</code></a></td><td><code>Merge tag '9.8.beta6' into t/34849/more_fixes_to_support_tox_4</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/6b8c666119fd96115b4efafc3b23984fdacf0f6b"><code>6b8c666</code></a></td><td><code>Merge #34839</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a2eb816f4f265d30e5f9e8b3721ee4c703a29e9b"><code>a2eb816</code></a></td><td><code>tox.ini: Require tox >= 4.2.7</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/516798af714b50af826969f4ca244f0366a781a5"><code>516798a</code></a></td><td><code>Merge #34849</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/d857fa05e6ce09f330b26eb5c56800cfd037004a"><code>d857fa0</code></a></td><td><code>pkgs/sagemath-categories: Add parts of sage.rings.polynomial</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`702ee8d`](https://github.com/sagemath/sagetrac-mirror/commit/702ee8d5a50971f0022079efeeafb3d522216e65)** to **[`d857fa0`](https://github.com/sagemath/sagetrac-mirror/commit/d857fa05e6ce09f330b26eb5c56800cfd037004a)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`d857fa0`](https://github.com/sagemath/sagetrac-mirror/commit/d857fa05e6ce09f330b26eb5c56800cfd037004a)** to **[`10d3767`](https://github.com/sagemath/sagetrac-mirror/commit/10d376757a552e2c03c5c14841fdab11c5605770)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:216"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/7f279d1f77c7c8402ef9bf337252b9176e13c26a"><code>7f279d1</code></a></td><td><code>pkgs/sagemath-polyhedra/tox.ini: Change to allowlist_externals</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/10d376757a552e2c03c5c14841fdab11c5605770"><code>10d3767</code></a></td><td><code>pkgs/sagemath-categories/setup.cfg.m4: Add to package_data</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:217"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c28996d9abeeaf94c1c083de9c7d6cdf23e83617"><code>c28996d</code></a></td><td><code>pkgs/sagemath-categories: Add parts of sage.sets</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`10d3767`](https://github.com/sagemath/sagetrac-mirror/commit/10d376757a552e2c03c5c14841fdab11c5605770)** to **[`c28996d`](https://github.com/sagemath/sagetrac-mirror/commit/c28996d9abeeaf94c1c083de9c7d6cdf23e83617)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`c28996d`](https://github.com/sagemath/sagetrac-mirror/commit/c28996d9abeeaf94c1c083de9c7d6cdf23e83617)** to **[`62da507`](https://github.com/sagemath/sagetrac-mirror/commit/62da50773da5c137b5265f43174f5ac4355edd1b)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:218"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/7b02fb249669c48ff7ae76da44e79a0693948749"><code>7b02fb2</code></a></td><td><code>sage.structure.element.NumberFieldElement: New ABC</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a9dacbe0f717ac312be98609228a936afaf8a4e3"><code>a9dacbe</code></a></td><td><code>is_NumberFieldElement: Deprecate, replace uses by isinstance</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/81e875bf78aa10539be5254db89cc93f51d47eec"><code>81e875b</code></a></td><td><code>Merge #34931</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/1cc09cccbde3b3eafb4f123d4f06ae8cb8cace28"><code>1cc09cc</code></a></td><td><code>Replace some imports of NumberFieldElement classes by ABC</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/4dfefbf9e0cfc6f4aa4a402b6a053ffe6e838792"><code>4dfefbf</code></a></td><td><code>src/sage/rings/number_field/number_field_element.pyx: Update doctest with deprecation warning</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/62da50773da5c137b5265f43174f5ac4355edd1b"><code>62da507</code></a></td><td><code>Merge #34931</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`62da507`](https://github.com/sagemath/sagetrac-mirror/commit/62da50773da5c137b5265f43174f5ac4355edd1b)** to **[`f084629`](https://github.com/sagemath/sagetrac-mirror/commit/f08462948e94cffb886973ea6cdd3b248cfaf7dd)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:219"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/f08462948e94cffb886973ea6cdd3b248cfaf7dd"><code>f084629</code></a></td><td><code>pkgs/sagemath-polyhedra/setup.cfg.m4: Allow python 3.11</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:220"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ba246ff3fab204c012e25077f8e40b840dbb894b"><code>ba246ff</code></a></td><td><code>src/sage/combinat/all__sagemath_polyhedra.py: New</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/58f7767e5b2be5c3c250a3aac0bc4e283c2a79f2"><code>58f7767</code></a></td><td><code>Use try..except when importing polynomial implementation classes</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`f084629`](https://github.com/sagemath/sagetrac-mirror/commit/f08462948e94cffb886973ea6cdd3b248cfaf7dd)** to **[`58f7767`](https://github.com/sagemath/sagetrac-mirror/commit/58f7767e5b2be5c3c250a3aac0bc4e283c2a79f2)**

**Comment by mkoeppe** [comment]
Changed dependencies from **#32779, #34855, #34849, #32709** to **#32779, #34855, #34849, #32709, #34849, #34931, #34941**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`58f7767`](https://github.com/sagemath/sagetrac-mirror/commit/58f7767e5b2be5c3c250a3aac0bc4e283c2a79f2)** to **[`8270697`](https://github.com/sagemath/sagetrac-mirror/commit/82706978f9bf86f9155fbc26d3d5bfd34105a8a6)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:222"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c9afec2374ce020a6f77753dd6b8fa43690aa340"><code>c9afec2</code></a></td><td><code>src/sage/rings/integer.pyx: Mark doctests # optional - sage.libs.pari, sage.rings.number_field, sage.symbolic</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a4f7955c5817d9de8e31e01aac8816501a4d48e4"><code>a4f7955</code></a></td><td><code>pkgs/sagemath-categories/MANIFEST.in.m4: Add sage.rings.finite_rings.integer_mod, sage.rings.factorint</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e46b5d578a7c8d724088518ad89100b021e99c35"><code>e46b5d5</code></a></td><td><code>src/sage/rings/polynomial/polynomial_element_generic.py: Use try...except for an unpickle_override import</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/f4730da4367957e401fe6b40366e7f3fae00495b"><code>f4730da</code></a></td><td><code>pkgs/sagemath-categories/MANIFEST.in.m4: Add sage.rings.finite_rings.integer_mod_ring</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/86f2f74bde939dfe4332be451f4b87f65212fff4"><code>86f2f74</code></a></td><td><code>src/sage/geometry/polyhedron: Add # optional - sage.combinat, sage.groups, sage.plot, sage.rings.number_field, sage.symbolic</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/9b6cb96b12ecacdf7780769cf54c4d7f045de6c2"><code>9b6cb96</code></a></td><td><code>pkgs/sagemath-polyhedra: Add sage.numerical.mip etc</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ad6a4663b24a1951c6992a750ae38f370c109335"><code>ad6a466</code></a></td><td><code>pkgs/sagemath-polyhedra/tox.ini: Add more passing files</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/131e3be3cfaa4ea77783d7e42705860402cf6971"><code>131e3be</code></a></td><td><code>pkgs/sagemath-polyhedra: Add sage.numerical.mip etc (fixup)</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/25f84e30e43cc5536504738a0e42e257237ebb22"><code>25f84e3</code></a></td><td><code>src/sage/numerical/backends/scip_backend.pyx: Remove unused imports</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/82706978f9bf86f9155fbc26d3d5bfd34105a8a6"><code>8270697</code></a></td><td><code>Merge #34941</code></td></tr></table>

**Comment by mkoeppe** [comment]
Removed the branch from the issue description; replaced by #35095

**Comment by orlitzky** [comment]
This is a WONTFIX for upstream Sage. If you need these separate distributions, please use https://github.com/passagemath/passagemath.
